tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -118,4 +195,182 @@ TEST_F(VersionedLayerClientTest, GetPartitions) {
   EXPECT_TRUE(mock_server_client_->Verify());
 }
 
+TEST_F(VersionedLayerClientTest, GetAggregatedData) {
+  olp::client::HRN hrn(kTestHrn);
+
+  constexpr auto kTileId = "23606938";
+  constexpr auto kLayer = "testlayer";
+  constexpr auto kQuadTreeDepth = 4;
+  const auto root_tile = olp::geo::TileKey::FromHereTile(kTileId);
+  const auto tile = root_tile.ChangedLevelTo(16);
+  const auto request = olp::dataservice::read::TileRequest().WithTileKey(tile);
+
+  // authentification not needed for the test
+  settings_->authentication_settings = boost::none;
+
+  {
+    SCOPED_TRACE("Requested tile");
+    const auto data_handle =
+        mockserver::DefaultResponses::GenerateDataHandle(tile.ToHereTile());
+    const auto data = mockserver::DefaultResponses::GenerateData();
+
+    {
+      mock_server_client_->MockGetResponse(
+          mockserver::DefaultResponses::GenerateApisResponse(kTestHrn));
+      mock_server_client_->MockGetResponse(
+          mockserver::DefaultResponses::GenerateVersionResponse(44));
+      mock_server_client_->MockGetResponse(
+          kLayer, root_tile,
+          GenerateQuadTreeResponse(root_tile, kQuadTreeDepth,
+                                   {1, 3, 12, 13, 14, 15, 16}));
+      mock_server_client_->MockGetResponse(kLayer, data_handle, data);
+    }
+
+    auto client =
+        std::make_unique<olp::dataservice::read::VersionedLayerClient>(
+            hrn, kLayer, boost::none, *settings_);
+
+    auto future = client->GetAggregatedData(request).GetFuture();
+    auto response = future.get();
+    const auto result = response.MoveResult();
+    const auto& result_data = result.GetData();
+
+    ASSERT_TRUE(response.IsSuccessful())
+        << response.GetError().GetMessage().c_str();
+    ASSERT_TRUE(result_data);
+    ASSERT_EQ(result_data->size(), data.size());
+    EXPECT_TRUE(std::equal(data.begin(), data.end(), result_data->begin()));
```

Comment:
Vector has comparison operators.

Reply:
yeah but `result_data` is a string and `data` is a vector, so the operator can't be used here.